packages/graphql/src/lib/schema.ts

```diff
@@ -0,0 +1,105 @@
+import {
+  type DecoratorContext,
+  type DecoratorFunction,
+  type Namespace,
+  type Program,
+  type Type,
+  getTypeName,
+  validateDecoratorUniqueOnNode,
+} from "@typespec/compiler";
+
+import { reportDiagnostic } from "../lib.js";
+import { useStateMap } from "./state-map.js";
+
+// This will set the namespace for all the decorators
+export const namespace = "TypeSpec.GraphQL";
```

Comment:
```suggestion
```
This should not be needed when using `export const $decorators`.

Reply:
Turns out we do need this. When we import `schema.js` into `schema.tsp`, the definition must be namespaced in order to match the TypeSpec declaration.

So we have two different paths for namespacing:
1. `export const $decorators` in `tsp-index.js` sets the namespace explicitly for all decorators when used by external code
2. `export const namespace = ` in `schema.js` sets the namespace when the `$schema` decorator is imported directly.

Since these are two independent import paths, they [should not conflict](https://discord.com/channels/1247582902930116749/1250119513681301514/1306717174551023636).Turns out we do need this. When we import `schema.js` into `schema.tsp`, the definition must be namespaced in order to match the TypeSpec declaration.

So we have two different paths for namespacing:
1. `export const $decorators` in `tsp-index.js` sets the namespace explicitly for all decorators when used by external code
2. `export const namespace = ` in `schema.js` sets the namespace when the `$schema` decorator is imported directly.

Since these are two independent import paths, they [should not conflict](https://discord.com/channels/1247582902930116749/1250119513681301514/1306717174551023636).